website/src/plugins/github-release-plugin.ts

```diff
@@ -0,0 +1,206 @@
+import type { Plugin } from '@docusaurus/types';
+
+// Type definition for an asset from the GitHub API response
+interface GitHubAsset {
+  name: string;
+  browser_download_url: string;
+}
+
+// Type definition for the relevant parts of the GitHub release API response
+interface GitHubRelease {
+  tag_name?: string;
+  name: string;
+  assets: GitHubAsset[];
+}
+
+// --- Data type definitions for each OS ---
+
+export interface LinuxDownloadData {
```

Comment:
suggestion(non-blocking): I like abstraction, just to mention that we could have an interface DownloadData with a field version, and the Linux & Windows & MacOs could extend it

Reply:
772ad719a26c59564f7e3c649f07028d56735ec4